### PR TITLE
feat: Add Radio, update Checkbox and Switch

### PR DESCRIPTION
**Pull request by alimpens**
This adds a Radio component. During development I ran into issues which also apply to Checkbox and Switch, so I've updated those as well. The most important change in that regard is a fix for invisible focus rings on older browsers that do not support `focus-visible` (e.g. Safari < 15.4).

### CODE REVIEW

File: packages/css/src/components/radio/radio.scss [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  &:focus + label {
+    outline: auto;
+  }
+
+  &:focus:not(:focus-visible) + label {
+    outline: 0;
+  }
+}
+
+.amsterdam-radio__circle {
```
Comment (VincentSmedinga) [inline packages/css/src/components/radio/radio.scss:25]: How about `.amsterdam-radio__control`?

File: packages/css/src/components/radio/radio.scss [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    background-image: var(--amsterdam-radio-circle-checked-background-image);
+    background-position: center;
+    background-repeat: no-repeat;
+    background-size: 1rem;
+  }
+}
+
+// Invalid unchecked
+.amsterdam-radio__input:invalid,
+.amsterdam-radio__input[aria-invalid="true"] {
```
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/css/src/components/radio/radio.scss:85]: `:invalid` is the native invalid state. [Radio buttons don't participate in constraint validation](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio#validation:~:text=Radio%20buttons%20don%27t%20participate%20in%20constraint%20validation) though, so we could get rid of that I think?

`aria-invalid` is used by libraries such as React hook form, to toggle invalid states for additional (non-native) validation.

File: packages/css/src/components/radio/radio.scss [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+  background-image: var(--amsterdam-radio-circle-disabled-checked-hover-background-image);
+}
+
+// Disabled invalid unchecked hover
+:is(.amsterdam-radio__input:invalid:disabled, .amsterdam-radio__input[aria-invalid="true"]:disabled)
+  + .amsterdam-radio__label:hover
+  .amsterdam-radio__circle::after {
+  // TODO: currently disabled invalid gets the same styling as disabled. This should get its own styling.
+  border-color: var(--amsterdam-radio-circle-disabled-border-color);
+}
```
[the remark this answers is not in the document]
Comment (alimpens) [reply packages/css/src/components/radio/radio.scss:167]: They do, technically, but it's conventional to style this the same as when not hovering. This rule is necessary to override the invalid hover state (without it, the circle becomes red on hover).

File: packages/css/src/components/radio/radio.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages/css/src/components/radio/radio.scss:77]: Thanks and yes, let’s remove this selector for radios then.

File: packages/css/src/components/radio/radio.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages/css/src/components/radio/radio.scss:159]: Alright, I can see how this may be the least complex approach.

**Review by VincentSmedinga — DISMISSED** [review]

**Review by VincentSmedinga — APPROVED** [review]

**Pull request merged**